autogpts/autogpt/autogpt/config/config.py

```diff
@@ -297,36 +301,54 @@ def build_config_from_env(cls, project_root: Path = PROJECT_ROOT) -> Config:
         return config
 
 
-def assert_config_has_openai_api_key(config: Config) -> None:
+async def assert_config_has_openai_api_key(config: Config) -> None:
     """Check if the OpenAI API key is set in config.py or as an environment variable."""
-    if not config.openai_credentials:
-        print(
-            Fore.RED
-            + "Please set your OpenAI API key in .env or as an environment variable."
-            + Fore.RESET
+    key_pattern = r"^sk-\w{48}"
+    openai_api_key = (
+        config.openai_credentials.api_key.get_secret_value()
+        if config.openai_credentials and config.openai_credentials.api_key
+        else ""
+    )
+
+    # If there's no credentials or empty API key, prompt the user to set it
+    if not openai_api_key:
+        logger.error(
+            "Please set your OpenAI API key in .env or as an environment variable."
+        )
+        logger.info(
+            "You can get your key from https://platform.openai.com/account/api-keys"
         )
-        print("You can get your key from https://platform.openai.com/account/api-keys")
-        openai_api_key = input(
-            "If you do have the key, please enter your OpenAI API key now:\n"
+        openai_api_key = await clean_input(
+            config, "If you do have the key, please enter your OpenAI API key now:"
         )
-        key_pattern = r"^sk-\w{48}"
         openai_api_key = openai_api_key.strip()
         if re.search(key_pattern, openai_api_key):
             os.environ["OPENAI_API_KEY"] = openai_api_key
             config.openai_credentials = OpenAICredentials(
                 api_key=SecretStr(openai_api_key)
             )
-            print(
-                Fore.GREEN
-                + "OpenAI API key successfully set!\n"
-                + Fore.YELLOW
-                + "NOTE: The API key you've set is only temporary.\n"
-                + "For longer sessions, please set it in .env file"
-                + Fore.RESET
+            logger.info(
+                "OpenAI API key successfully set!",
+                extra={"color": Fore.GREEN},
+            )
+            logger.info(
+                "NOTE: The API key you've set is only temporary. "
+                "For longer sessions, please set it in .env file",
+                extra={"color": Fore.YELLOW},
```

Comment:
Why `print` -> `logger.info`?

Reply:
Logging is the application reporting what's going on. If a user wants less info, they can raise the log level and e.g. only get `WARNING` and higher urgency messages.

`print()` is direct terminal output to interface with the user. This is part of the UI, functionally different from the logging output.